src/c_api/c_api.cc

```diff
@@ -322,6 +322,38 @@ int MXNDArrayLoad(const char* fname,
   API_END();
 }
 
+int MXNDArrayLoadFileContent(const void *nd_file,
+                            size_t size,
+                            mx_uint *out_size,
+                            NDArrayHandle** out_arr,
+                            mx_uint *out_name_size,
+                            const char*** out_names) {
+  MXAPIThreadLocalEntry *ret = MXAPIThreadLocalStore::Get();
+  ret->ret_vec_str.clear();
+  API_BEGIN();
+  std::vector<NDArray> data;
+  std::vector<std::string> &names = ret->ret_vec_str;
+  {
+    std::unique_ptr<dmlc::MemoryFixedSizeStream> fi(new dmlc::MemoryFixedSizeStream((void*)nd_file, size)); // NOLINT(*)
+    mxnet::NDArray::Load(fi.get(), &data, &names);
```

Comment:
what happens if there is an error/formatting problem? from the python perspective? how catastrophic is that?

Reply:
Just checked multiple versions of this, it will return -1 with the last error message being:
`Check failed: header == kMXAPINDArrayListMagic Invalid NDArray file format`
Python can check for that pretty easily.

Just added a check against NULL pointers which I didn't think of originally.